### PR TITLE
Add status messages for client errors

### PR DESCRIPTION
**Pull request by cgutman**
## Description
We weren't sending any status messages to clients with our failure responses. As a result, they'd just see a 400/404/503 error code with no context to explain what the error actually was.

### Screenshot
IMG_0539

image

image

## Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/nvhttp.cpp [hunk 1]
```diff
@@ -842,6 +830,7 @@ namespace nvhttp {
       if (video::probe_encoders()) {
         tree.put("root.resume", 0);
         tree.put("root.<xmlattr>.status_code", 503);
+        tree.put("root.<xmlattr>.status_message", "Failed to initialize video capture/encoding. Is a display connected and turned on?");
```
Comment (ReenigneArcher) [inline src/nvhttp.cpp:833]:
```suggestion
        tree.put("root.<xmlattr>.status_message", "Failed to initialize video capture/encoding. Is a display connected to the selected adapter, and turned on?");
```

This might make it too wordy, but I think a lot of users don't understand the display needs to be connected to the GPU.

**Review by ReenigneArcher — COMMENTED** [review]
This will be a nice improvement.

File: src/nvhttp.cpp (hunk 1 above)
Comment (cgutman) [reply src/nvhttp.cpp:833]: Most users probably won't be selecting adapters (and I agree it might be too wordy). I'll put a message next to the encoder and adapter selection fields in the UI instead.
